fe/fe-core/src/main/java/org/apache/doris/datasource/paimon/source/PaimonSource.java

```diff
@@ -36,7 +37,8 @@ public class PaimonSource {
     public PaimonSource(TupleDescriptor desc) {
         this.desc = desc;
         this.paimonExtTable = (PaimonExternalTable) desc.getTable();
-        this.originTable = paimonExtTable.getPaimonTable();
+        this.originTable = paimonExtTable.getPaimonTable(
+                ConnectContext.get().getStatementContext().getSnapshot(paimonExtTable));
```

Comment:
do we need check `ConnectContext.get() == null` and `ConnectContext.get().getStatementContext()`. sometimes they do not exist